misc,tests: Add dummy jobs to workflows for status checks

BobbyRBruce opened the pull request:
With this change we can block PRs if the compiler, daily, and/or weekly tests are failing by setting these dummy jobs as require status checks.

BobbyRBruce committed:
misc,tests: Add dummy jobs to workflows for status checks

Change-Id: I52e42b6f93cfbb1a8e4800a3f6e264d49bebb06c

powerjg reviewed (commented):
Probably shouldn't run this on self hosted

BobbyRBruce committed:
tests,misc: Update CI Tests 'testlib-quick' runs-on

Here it's more sensible to use a GitHub hosted runner. This job is
miniscule and is used to check the other tests have completed
successfully. It makes sense for this not to be on our own self-hosted
runner.

Change-Id: I5377e025334d43eaedd0fc61e5c708ba61255d28

BobbyRBruce commented:
> Probably shouldn't run this on self hosted

Fixed.
